libr/bin/p/bin_mach0.c

```diff
@@ -213,7 +213,7 @@ static RList *symbols(RBinFile *bf) {
 	Sdb *symcache = sdb_new0 ();
 	bin = (struct MACH0_(obj_t) *) obj->bin_obj;
 	for (i = 0; !syms[i].last; i++) {
-		if (!syms[i].name[0] || syms[i].addr < 100) {
+		if (!syms[i].name || syms[i].addr < 100) {
```

Comment:
why this change? I think it slightly changes the behaviour.

Reply:
When symbols were allocated on the stack `!syms[i].name[0]` was a way of telling that the symbol is an empty string. Now, that the symbols are allocated on the heap, we don't have empty strings - but NULL pointers. In the case where `syms[i].name == NULL`, wouldn't `syms[i].name[0]` result in a segfault?

Reply:
yes, but can you have `syms[i] != NULL && syms[i].name[0] == '\0'`?